contribute/design/1.x/cstor_cr_schema/apis.md

```diff
@@ -0,0 +1,558 @@
+
+
+- CSPC can be used to specify tolerations for pool pods and there is a defaulting mechanism too. Please refer to the following PR to understand more.
+https://github.com/openebs/maya/pull/1549
+
+
+- CSPC can be used to do block device replacement.
+
+
+- CSPC can be used to do pool expansion.
+
+
+- CSPC can be used to create a pool on a new brought up node. ( Horizontal scale )
+
+
+Following is the current CSPC schema in go struct : 
+
+```go
+// CStorPoolCluster describes a CStorPoolCluster custom resource.
+type CStorPoolCluster struct {
+  metav1.TypeMeta   `json:",inline"`
+  metav1.ObjectMeta `json:"metadata,omitempty"`
+  Spec              CStorPoolClusterSpec   `json:"spec"`
+  Status            CStorPoolClusterStatus `json:"status"`
+  VersionDetails    VersionDetails         `json:"versionDetails"`
+}
+
+// CStorPoolClusterSpec is the spec for a CStorPoolClusterSpec resource
+type CStorPoolClusterSpec struct {
+  // Pools is the spec for pools for various nodes
+  // where it should be created.
+  Pools []PoolSpec `json:"pools"`
+  // DefaultResources are the compute resources required by the cstor-pool
+  // container.
+  // If the resources at PoolConfig is not specified, this is written
+  // to CSPI PoolConfig.
+  DefaultResources *corev1.ResourceRequirements `json:"resources"`
+  // AuxResources are the compute resources required by the cstor-pool pod
+  // side car containers.
+  DefaultAuxResources *corev1.ResourceRequirements `json:"auxResources"`
+  // Tolerations, if specified, are the pool pod's tolerations
+  // If tolerations at PoolConfig is empty, this is written to
+  // CSPI PoolConfig.
+  Tolerations []corev1.Toleration `json:"tolerations"`
+
+  // DefaultPriorityClassName if specified applies to all the pool pods
+  // in the pool spec if the priorityClass at the pool level is
+  // not specified.
+  DefaultPriorityClassName string `json:"priorityClassName"`
+}
+
+//PoolSpec is the spec for pool on node where it should be created.
+type PoolSpec struct {
+  // NodeSelector is the labels that will be used to select
+  // a node for pool provisioning.
+  // Required field
+  NodeSelector map[string]string `json:"nodeSelector"`
+  // RaidConfig is the raid group configuration for the given pool.
+  RaidGroups []RaidGroup `json:"raidGroups"`
+  // PoolConfig is the default pool config that applies to the
+  // pool on node.
+  PoolConfig PoolConfig `json:"poolConfig"`
+}
+
+// PoolConfig is the default pool config that applies to the
+// pool on node.
+type PoolConfig struct {
+  // Cachefile is used for faster pool imports
+  // optional -- if not specified or left empty cache file is not
+  // used.
+  CacheFile string `json:"cacheFile"`
+  // DefaultRaidGroupType is the default raid type which applies
+  // to all the pools if raid type is not specified there
+  // Compulsory field if any raidGroup is not given Type
+  DefaultRaidGroupType string `json:"defaultRaidGroupType"`
```

Comment:
@vishnuitta should we really need to ask user to give persistent storage path to store sock file?

Reply:
sock file can be on emptyDir